crates/snp_validation/src/lib.rs

```diff
@@ -288,94 +310,129 @@ impl Snp {
+        if let Some(config) = config {
+            let config = &config.config;
+            if !config.minimum_abi.matches(&report.body.policy.into()) {
+                bail!("snp minimum abi not met");
             }
-        }
 
-        if report.body.vmpl != 0 {
-            return Err(anyhow!("snp report vmpl field not set by Enarx"));
-        }
+            if !config.enarx_signer.is_empty() {
+                let mut signed = false;
+                for signer in &config.enarx_signer {
+                    if report.body.author_key_digest == signer.as_ref() {
+                        signed = true;
+                        break;
+                    }
+                }
+                ensure!(signed, "snp untrusted enarx author_key");
+            }
 
-        if report.body.guest_svn != 0 {
-            return Err(anyhow!("snp report guest_svn field not set by Enarx"));
-        }
+            if !config.id_key_digest.is_empty() {
+                let mut signed = false;
+                for signer in &config.id_key_digest {
+                    if report.body.id_key_digest == signer.as_ref() {
+                        signed = true;
+                        break;
+                    }
+                }
+                ensure!(signed, "snp untrusted enarx id_key");
+            }
 
-        // Check field set by Enarx
-        for value in report.body.report_id_ma {
-            if value != 255 {
-                return Err(anyhow!(
-                    "snp report report_id_ma field not the value set by Enarx"
-                ));
+            if !config.id_key_digest_blacklist.is_empty() {
+                for signer in &config.id_key_digest_blacklist {
+                    if report.body.id_key_digest == signer.as_ref() {
+                        bail!("snp untrusted enarx id_key");
+                    }
+                }
+            }
+
+            if !config.enarx_hash.is_empty() {
+                let mut allowed = false;
+                for hash in &config.enarx_hash {
+                    if report.body.measurement == hash.as_ref() {
+                        allowed = true;
+                        break;
+                    }
+                }
+                ensure!(allowed, "snp untrusted enarx hash");
+            }
+
+            if let Some(flags) = config.platform_info_flags {
+                if flags != report.body.plat_info.flag {
+                    bail!("snp unexpected memory mode");
+                }
             }
         }
 
         if !dbg {
             // Validate that the certification request came from an SNP VM.
-            let hash = Sha384::digest(&cri.public_key.to_vec()?);
+            let hash = Sha384::digest(cri.public_key.to_vec()?);
             if hash.as_slice() != &report.body.report_data[..hash.as_slice().len()] {
-                return Err(anyhow!("snp report.report_data is invalid"));
```

Comment:
if we're refactoring, this should use `ensure!`

Reply:
Fixed